kms-message/src/kms_crypto_libcrypto.c

```diff
@@ -91,4 +91,48 @@ kms_sha256_hmac (void *unused_ctx,
                 NULL) != NULL;
 }
 
+bool
+kms_sign_rsaes_pkcs1_v1_5 (void *unused_ctx,
+                           const char *private_key,
+                           size_t private_key_len,
+                           const char *input,
+                           size_t input_len,
+                           unsigned char *signature_out)
+{
+   EVP_MD_CTX *ctx;
+   EVP_PKEY *pkey = NULL;
+   bool ret = false;
+   size_t signature_out_len;
+
+   ctx = EVP_MD_CTX_new ();
+   pkey = d2i_PrivateKey (EVP_PKEY_RSA,
+                          NULL,
+                          (const unsigned char **) &private_key,
+                          private_key_len);
+   if (!pkey) {
+      goto cleanup;
+   }
+
+   ret = EVP_DigestSignInit (ctx, NULL, EVP_sha256 (), NULL /* engine */, pkey);
```

Comment:
Looks like the default is RSA_PKCS1_PADDING which made this really easy.

Reply:
Yeah, if we needed a different padding, I think I could set the padding type as a property of the EVP_PKEY with `EVP_PKEY_CTX_set_rsa_padding`.